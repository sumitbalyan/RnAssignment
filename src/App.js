import React, { useState } from 'react'
import { View, StyleSheet, SafeAreaView, Button, FlatList } from 'react-native'
import Items from './Items';
import Loader from './Loader';
import Search from './Search';
import { getData } from './services.js';

const numColumns = 4;

const App = ({navigation}) => {
    const [state, setstate] = useState({
        indicator: false,
        data: [],
        search: '',
    });
    
    const setSeachData = (search)=> {
        setstate({...state, search});
    }
    const onSearch = async ()=> {
        const {search} = state;
        if(search != ''){
        setstate({...state, indicator: true});
        const resultData = await getData(search); 
        const data = resultData.results;
        setstate({...state, indicator: false, data});
        }
    }
    const showDetails = (item)=> {
        navigation.navigate('Details', {item})
    }
    const renderItems = ({ item }) => {
        return(
            <Items
                item={item}
                onPress={() => showDetails(item)}
                numColumns={numColumns}
          />
      )
    }

    return (
        <View style={styles.container}>
             <SafeAreaView>
                <Loader loading = {state.indicator} />
                <View style={styles.searchContainer}>
                    <Search setSeachData = {setSeachData} />
                    <Button title='GO' onPress={onSearch} />
                </View>
                <FlatList
                    numColumns={numColumns}
                    data={state.data}
                    renderItem={renderItems}
                    keyExtractor={(item)=> item.trackId}
            />
            </SafeAreaView>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        marginTop : 10,
    },
    searchContainer: {
        flexDirection: 'row',
        marginLeft: 5,
    },
    item: {
        backgroundColor: 'red',
        padding: 10,
        marginVertical: 8,
        marginHorizontal: 16,
    },
    text: {
        color: 'white',
        fontSize: 16,
    },
})
export default App
