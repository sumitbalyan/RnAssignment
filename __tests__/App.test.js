import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/App';

test('Contact renders correctly', () => {
  const render = renderer.create(<App />).toJSON();
  expect(render).toMatchSnapshot();
});
