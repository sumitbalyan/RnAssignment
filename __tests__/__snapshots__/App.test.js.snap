// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 10,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000040",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-around",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "display": "flex",
              "height": 100,
              "justifyContent": "space-around",
              "width": 100,
            }
          }
        >
          <ActivityIndicator
            animating={false}
            color="#000"
            hidesWhenStopped={true}
            size="large"
          />
        </View>
      </View>
    </Modal>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginLeft": 5,
        }
      }
    >
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Search"
          placeholderTextColor="rgba(255,255,255,0.7)"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.35)",
              "borderRadius": 25,
              "color": "rgba(255,255,255,0.7)",
              "fontSize": 16,
              "height": 45,
              "marginHorizontal": 10,
              "paddingLeft": 45,
              "width": 650,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            GO
          </Text>
        </View>
      </View>
    </View>
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
